### PR TITLE
[CODE CAMP] BaseHardwareObjects

### PR DESCRIPTION
**Pull request by rhfogh**
Documented relevant classes

Also added fixes to satisfy pylint, and a few TODO suggestions (hence additional files)

### CODE REVIEW

File: BaseHardwareObjects.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 @enum.unique
 class HardwareObjectState(enum.Enum):
+    """Enumeration of ommon states, shared between all HardwraeObjects"""
```
Comment (IvarsKarpics) [inline BaseHardwareObjects.py:16]: common, HardwareObjects

File: BaseHardwareObjects.py [hunk 2]
```diff
@@ -706,17 +785,38 @@ def getDevice(self, deviceName):
...
+    And that use looks like it could be replaced by something else"""
+
     pass
 
 
 class Equipment(HardwareObject, DeviceContainer):
+    """Equipment class -old style
+
+    NB This class needs refactoring. Since many (soon: all??) contained
+     objects are no longer of class Devicem the code in here is unlikely to work."""
```
Comment (marcus-oscarsson) [inline BaseHardwareObjects.py:818]: typo Device ...

File: BaseHardwareObjects.py [hunk 3]
```diff
@@ -600,43 +659,53 @@ def __getattr__(self, attr):
...
         from .HardwareRepository import getHardwareRepository
 
         return getHardwareRepository().xml_source[self.name()]
 
 
 class HardwareObjectYaml(ConfiguredObject, HardwareObjectMixin):
-    """Yaml-configured hardware object"""
+    """Yaml-configured hardware object.
+
+    For use when we move confiugratoin out of xml and into yaml.
```
Comment (marcus-oscarsson) [inline BaseHardwareObjects.py:705]: confiugratoin -> configuration

File: BaseHardwareObjects.py [hunk 4]
```diff
@@ -468,47 +488,62 @@ def update_state(self, state=None):
...
 
     # Moved from HardwareObjectNode
     def clear_gevent(self):
-        """Clear gevent tasks. Override in subclasses
-
-        NB Not sure why this is needed, except maybe for object reloading?
+        """Clear gevent tasks, called when disconnecting a HardwareObject.
 
+        Override in subclasses as needed.
+i
```
Comment (marcus-oscarsson) [inline BaseHardwareObjects.py:531]: oops :)

File: BaseHardwareObjects.py [hunk 5]
```diff
@@ -657,6 +728,16 @@ def GUI(self, parent):
 
 
 class Device(HardwareObject):
+    """Old superclass for devices
```
Comment (marcus-oscarsson) [inline BaseHardwareObjects.py:731]: We should maybe as one of the lasts steps of all this, remove all usage of Device and Equipment and used HardwareObject instead.

**Comment by marcus-oscarsson** [comment]
Looks good to me, thanks !

File: BaseHardwareObjects.py (hunk 5 above)
Comment (rhfogh) [reply BaseHardwareObjects.py:731]: Seconded!

Comment (rhfogh) [reply BaseHardwareObjects.py:731]: @marcus-oscarsson Actually I would love to get rid of those. Do you think I could/should do it now, leaving the current Equipment instances a bit dangling?Or even roughly refactoring them? Or might it it be better to let people refactor those first and remove Equipment after that?

**Pull request merged**